### PR TITLE
Implement AddNorm

### PR DESCRIPTION
**Pull request by jackfreem**
## Description
- Implements AddNorm with some unit tests
- Restrucutres the test directory to not include "src," this was causing import issues for me & I figured this minimizes risk of having to constantly change our pythonpaths.

## Checklist

Please ensure the following before submitting the PR:

- [x] PR is Squashed
- [x] I have added/updated relevant documentation (e.g., README.md, inline docstrings).
- [x] I have written tests for my changes.
- [x] All tests pass locally.

## How to Test
```
cd frillsformer
pytest
```

### CODE REVIEW

File: src/layers/add_norm.py [hunk 1]
```diff
@@ -2,8 +2,39 @@
...
+
+        Args:
+            x: Input tensor from the previous layer
+            sublayer_output: Output tensor from the current sublayer
+
+        Returns:
+            torch.Tensor: Normalized tensor after adding x & sublayer_output
+        """
+        # Add residual connection and normalize
+        return self.norm(x + sublayer_output)
```
Comment (bradford-gill) [inline src/layers/add_norm.py:40]: in hind site, we def did not need an entire class just for this lol

File: tests/layers/test_add_norm.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        # Backward pass
+        loss = output.sum()  # Simple scalar loss
+        loss.backward()
+
+        # Check gradients
+        self.assertIsNotNone(input_tensor.grad)
+        self.assertIsNotNone(sublayer_output.grad)
+        self.assertFalse(torch.isnan(input_tensor.grad).any())
+        self.assertFalse(torch.isnan(sublayer_output.grad).any())
```
Comment (bradford-gill) [inline tests/layers/test_add_norm.py:73]: this is great work

File: pytest.ini [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[pytest]
```
Comment (bradford-gill) [inline pytest.ini:1]: nice

**Review by bradford-gill — APPROVED** [review]
tests are running. Great work king, this is quality stuff.

File: src/layers/add_norm.py (hunk 1 above)
Comment (jackfreem) [reply src/layers/add_norm.py:40]: I think it's good for the modularity! But yea, it was a bit funny when I realized the class one essentially a 1 liner.

**Pull request merged**